### ISSUE TITLE
Build fails on Linux 4.13.0

### ISSUE THREAD

**Issue by aiwenar** [gimli-rs/findshlibs#19]
Environment:

```
$ uname -srvmpio 
Linux 4.13.0-38-generic #43~16.04.1-Ubuntu SMP Wed Mar 14 17:48:43 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

Error:

```
$ cargo run --example list_segments
   Compiling findshlibs v0.4.0 (file://$SOURCE)
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "$HOME/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.1234nblqyzxw0hek.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.159zrtzhlq4sp7cu.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.181cuta0v63atwcm.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.1im38lueib99jsk0.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.1y16o1qfye96o7m0.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.233vi12nrzsr2rv9.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.2jqywn86b2gsqohu.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.3171x0bwu82dptu7.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.3ayaeypdcro9d6yk.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.3bfrbavjone0rmbn.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.3os1glt0smvmfmyt.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.3rngp6bm2u2q5z0y.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.49a7n47po4ttqjl7.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.49lx1q7cxvpykyv0.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.4a6dgwgsyclvnmy1.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.4bwka44cn8t3u4h3.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.4oc10dk278mpk1vy.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.4xq48u46a1pwiqn7.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.4yh8x2b62dcih00t.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.4ypvbwho0bu5tnww.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.8xzrsc1ux72v29j.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.98g0d9x8aw3akpe.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.9elsx31vb4it187.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.c6lbtaiefvx3wya.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.mxf6cedk2mb0uti.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.oa3rad818d8sgn4.rcgu.o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.y08g5q2x813c4wx.rcgu.o" "-o" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7" "$SOURCE/target/debug/examples/list_segments-2fce113d824486e7.crate.allocator.rcgu.o" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-nodefaultlibs" "-L" "$SOURCE/target/debug/deps" "-L" "$HOME/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "$SOURCE/target/debug/deps/libfindshlibs-331f6de4d90fa2b1.rlib" "$SOURCE/target/debug/deps/libcfg_if-61f98562e8c75752.rlib" "-Wl,--start-group" "$HOME/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-ab468c29abdca826.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-071d11d9595c3993.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc_jemalloc-3d5f5190cf76a5a9.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-eaf3fcc485d42fbe.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc_system-0c9cb96866fcd97d.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-50836b259641df0c.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-597b8d641e8320a3.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-265d56b5f34cf645.rlib" "-Wl,--end-group" "$HOME/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-692abfbca9bed82d.rlib" "-Wl,-Bdynamic" "-l" "dl" "-l" "rt" "-l" "pthread" "-l" "pthread" "-l" "gcc_s" "-l" "c" "-l" "m" "-l" "rt" "-l" "pthread" "-l" "util" "-l" "util"
  = note: $SOURCE/target/debug/examples/list_segments-2fce113d824486e7.233vi12nrzsr2rv9.rcgu.o: In function `_$LT$findshlibs..linux..SharedLibrary$LT$$u27$a$GT$$u20$as$u20$findshlibs..SharedLibrary$GT$::each::h2bc69c50c71f6242':
          $SOURCE/src/linux/mod.rs:179: undefined reference to `dl_iterate_phdr'
          collect2: error: ld returned 1 exit status
          

error: aborting due to previous error

error: Could not compile `findshlibs`.
```

**Comment by fitzgen** [comment]
> undefined reference to `dl_iterate_phdr'

wat. It is like glibc isn't being linked or something...

**Comment by aiwenar** [comment]
It would seem the generated binding tries to use `dl_iterate_phdr` as a C++ symbol:

```
$ rg dl_iterate_phdr target/debug/build/findshlibs-f4923a9e94804948/out/linux_bindings.rs
1067:    #[link_name = "\u{1}_Z15dl_iterate_phdr"]
1068:    pub fn dl_iterate_phdr(
$ readelf -s target/debug/examples/list_segments-2fce113d824486e7.233vi12nrzsr2rv9.rcgu.o | rg dl_iterate_phdr
    19: 0000000000000000     0 NOTYPE  GLOBAL DEFAULT  UND _Z15dl_iterate_phdr
```

compared to example (from `man dl_iterate_phdr`) compiled with clang:

```
$ readelf -s test.o | rg dl_iterate_phdr
     8: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND dl_iterate_phdr@GLIBC_2.2.5 (2)
    63: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND dl_iterate_phdr@@GLIBC_2.
```

The symbol is defined in `/lib/x86_64-linux-gnu/libc.so.6`:

```
$ readelf -s /lib/x86_64-linux-gnu/libc.so.6 | rg dl_iterate_phdr       
   316: 0000000000142ff0   465 FUNC    WEAK   DEFAULT   13 dl_iterate_phdr@@GLIBC_2.2.5
```

**Comment by philipc** [comment]
I think this was fixed by #27
